### ISSUE TITLE
Liberalise the handling of bar lines

### ISSUE THREAD

**Issue by newlandsvalley** [newlandsvalley/purescript-abc-parser#29]
The ABC specification is really messy here.  After listing the various possibilities of thick/thin line combinations and showing that only the single thin line can have repeat markers against it, it throws it all away by saying:

```
Abc parsers should be quite liberal in recognizing bar lines. In the wild, bar lines may have any shape, using a sequence of
 | (thin bar line), [ or ] (thick bar line), and : (dots), e.g. |[| or [|::: 
```
We need to improve things in order to accommodate the ::|  or |:: multiple repeat marker.  I think that the best we can do is firstly restrict ourselves to these bar line combinations:

```
   |    thin
   ||   thin-thin
   [|   thick-thin
   |]   thin-thick
   ]|   thick-thin
````

and then to allow any number up to a sensible maximum (0-2?) of repeat ```:``` markers both immediately before and after the bar lines, and also to allow the freestanding ```::``` version as a synonym for ```:|:```.  Note that it would probably be tricky also to include the ```|[``` combination because of the possibility of ambiguity when we come to adding variant markers.  Here, as far as I can see, the following would all be legal:

```
   |1 
   [1
   | [1
```

I guess this construct is best parsed as a separate entity immediately after the bar line and colon(s).

As fart as the ABC ADT is concerned, we retain the ```Begin/End/BeginAndEnd``` repeat descriptor but add to an integer indicating the number of repeats.  (Currently it is only 1). Otherwise, no change.

**Comment by newlandsvalley** [comment]
This has been fixed in the parser by 97c5da50d2281adf0a5996d40447bf3b62141359 but it still builds the old syntax tree.  On reflection, I think we should now change the description of a Bar line type to:  

```
type BarLine =
    { endRepeats :: Int
    , thickness :: Thickness
    , startRepeats :: Int
    , iteration :: Maybe Volta
    }
```
and get rid of Repeat altogether.  There is no repeat if the count is 0.

**Comment by newlandsvalley** [comment]
Fixed by 0bfc47922db4cf7ca833b714a97b368babf12550 in the voltas branch
